### DISCUSSION TITLE
Best way to overwrite the name of a resource.

### DISCUSSION THREAD

**Discussion by SamHep0803** [filamentphp/filament#3509] [Help] [answered]
Hi, I've got a model named "vACC", while this may not be the typical naming convention, it's what I've gone with. I've created a resource for this model but it has set the label as "V A C C", is there any way to disable this autocapitalization and word splitting? I understand that some of this can be set using $label and $pluralLabel but this seems a little unreliable and does not update the header from what I can see.

Thanks,
Sam.
<img width="535" alt="image" src="https://user-images.githubusercontent.com/14109339/184641107-8bed5103-a90a-4f49-b629-cca0715f6929.png">

**Comment by haztakki** [comment]
You can override it in the resource.

`public static ?string $label = 'Foo';`

https://filamentphp.com/docs/2.x/admin/resources/getting-started#model-labels

**Edit:** I just saw you mentioned $label already, I'm not sure how else you can do it. You could add a method on the base resource that dynamically sets the label. I would just stick to the default naming conventions though.

**Reply by SamHep0803** [reply]
If you had read the entirety of my original message, I said that whilst $label did fix *some* things, it does not completely replace everything, the breadcrumbs and title still stay the same as you can see in the screenshot attached.

<img width="273" alt="image" src="https://user-images.githubusercontent.com/14109339/184644781-490d83b8-0420-4920-8008-d04fb674250a.png">

**Reply by SamHep0803** [reply]
Is there really no way to overwrite the title and breadcrumb though? That seems a little off, in my opinion, the only reason I wanted to use the acronym is that the entire string would be "Virtual Area Control Center" which is just too long.

**Comment by ousid** [comment]
What about `$navigationLabel`, but it will change the navigation name, not the name on the `breadcrumb` or the title

```php
protected static ?string $navigationLabel = 'foo';
```

Edit: you can use `$breadcrumb` also to change the title:

```php
protected static ?string $breadcrumb = 'foo';
```

**Reply by SamHep0803** [reply]
Thanks for this, this when I pair this with $modelLabel and $pluralModelLabel it helps slightly but it's the title and the breadcrumb that still needs fixing.

**Reply by ousid** [reply]
What about `$breadcrumb` property?

**Reply by SamHep0803** [reply]
Ahhhhhhh, this will fix it, thanks!

**Reply by ousid** [reply]
Glade it got fixed! You are very welcome @SamHep0803 :)

**Comment by SamHep0803** [comment, marked as answer]
Ok using a mix of `$modeLabel`, `$pluralModelLabel`, `$breadcrumb`, `$navigationLabel`, and manually setting the `$title` of pages, I have managed to update everything to their intended values. I do think that there should be some kind of way to disable this autocapitalization for specific resources, however. Thank you @ousid and @HazJ for all your help :)

**Comment by kurniawan750** [comment]
hello,
i have problem to renaming header in v3 documentation :
`public static function getHeading(): string
     {
         return 'Test Custom Header';
     }`
it not change, then i use :
`public static ?string $label = 'Foo';`
it work using v2 but header become Foos with s behind it, how to remove s ?
i can manage to name with `public static function getBreadcrumb(): string`
thanks.

**Reply by ousid** [reply]
Use `$modelLabel` instead of `$label`

```php
public static function getModelLabel()
{
 return '...';
}
```

**Reply by kurniawan750** [reply]
thank you for reply, i can managed with comment above using :
`    public static ?string $pluralModelLabel = 'Database 2022';
`

**Comment by AliKar97** [comment]
just add the following at the top of your resource class to get rid of the extra "s" character:

`protected static ?string $pluralLabel = 'yourName';
`
